vendor/engines/split_accounts/app/services/split_accounts/remainder_applier.rb

```diff
@@ -0,0 +1,47 @@
+module SplitAccounts
+
+  class RemainderApplier
+
+    attr_reader :original, :split_objects, :splits
+
+    # split_objects and splits must have the same order
+    def initialize(original, split_objects, splits)
+      raise "There should be the same number of split objects as splits" if splits.size != split_objects.size
+      @original = original
+      @split_objects = split_objects
+      @splits = splits
+    end
+
+    def apply_remainders(attributes)
+      Array(attributes).each do |attribute|
+        apply_remainder(attribute)
+      end
+    end
+
+    def apply_remainder(attribute)
+      total_value = original.public_send(attribute)
+      return if total_value.blank?
+
+      remainder = total_value - floored_total(attribute)
+
+      new_value = object_to_receive_remainder.public_send(attribute) + remainder
+      object_to_receive_remainder.public_send("#{attribute}=", new_value)
+    end
+
+    private
+
+    def object_to_receive_remainder
+      split_objects[remainder_index]
+    end
+
+    def remainder_index
+      @remainder_index ||= splits.find_index(&:extra_penny?)
+    end
+
+    def floored_total(attribute)
+      split_objects.reduce(BigDecimal(0)) { |sum, item| sum + item.public_send(attribute) }
```

Comment:
I'm not completely on-board with `Style/SingleLineBlockParams` imposing variable names for one-line `reduce`. But the line is long enough to become a 3-line `do`/`end` block, which should clear up the Code Climate :x:.